### PR TITLE
[Package Signing] Improved CertificateVerificationResult type

### PR DESCRIPTION
**Pull request by loic-sharma**
This is a subset of https://github.com/NuGet/NuGet.Jobs/pull/309, meant to reduce the number of changes in that PR.

### CODE REVIEW

File: src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs [hunk 1]
```diff
@@ -13,27 +13,139 @@ namespace Validation.PackageSigning.ValidateCertificate
...
+            DateTime? revocationTime = null)
+        {
+            if (status != EndCertificateStatus.Revoked && revocationTime.HasValue)
+            {
+                throw new ArgumentException(
+                    $"End certificate revoked at {revocationTime} but status isn't {nameof(EndCertificateStatus.Revoked)}",
+                    nameof(status));
+            }
+
+            switch (status)
```
Comment (dtivel) [inline src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs]: Considering throwing for the `default` case.

File: src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs [hunk 2]
```diff
@@ -13,27 +13,139 @@ namespace Validation.PackageSigning.ValidateCertificate
...
         /// </summary>
-        public X509ChainStatusFlags StatusFlags { get; set; }
+        public X509ChainStatusFlags StatusFlags { get; }
 
         /// <summary>
         /// The time that the end <see cref="X509Certificate2"/>'s status was last updated, according to the
-        /// Certificate Authority. If <see cref="Status"/> is <see cref="EndCertificateStatus.Revoked"/>
-        /// or <see cref="EndCertificateStatus.Unknown"/>, this will have a value of <c>null</c>.
+        /// Certificate Authority. This value will be <c>null</c> if the <see cref="Status"/> is
+        /// <see cref="EndCertificateStatus.Unknown"/> or if the status could not be determined.
```
Comment (dtivel) [inline src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs]: Should you enforce this rule in the constructor like you enforce the rule for revoked below?

File: tests/Validation.PackageSigning.ValidateCertificate.Tests/CertificateVerificationResultFacts.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
+﻿using System;
```
Comment (dtivel) [inline tests/Validation.PackageSigning.ValidateCertificate.Tests/CertificateVerificationResultFacts.cs]: Add license header.  Also check other files in this PR.

File: src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs (hunk 2 above)
Comment (loic-sharma) [reply src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs]: This isn't intended to be strictly enforced, I will reword this.

File: src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs (hunk 1 above)
Comment (loic-sharma) [reply src/Validation.PackageSigning.ValidateCertificate/CertificateVerificationResult.cs]: Added

File: tests/Validation.PackageSigning.ValidateCertificate.Tests/CertificateVerificationResultFacts.cs (hunk 3 above)
Comment (loic-sharma) [reply tests/Validation.PackageSigning.ValidateCertificate.Tests/CertificateVerificationResultFacts.cs]: Fixed

**Pull request merged**